api/v1/testkube.yaml

```diff
@@ -65,89 +65,155 @@ paths:
       tags:
         - executions
         - api
-      summary: 'Get script execution by ID'
-      description: 'Returns execution with given executionID'
+      summary: "Get script execution by ID"
+      description: "Returns execution with given executionID"
       operationId: getExecutionByID
       responses:
         200:
           description: successful operation
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Execution'
+                $ref: "#/components/schemas/Execution"
         500:
-          description: 'problem with getting script executions from storage'
+          description: "problem with getting script executions from storage"
           content:
             application/problem+json:
               schema:
                 type: array
                 items:
-                  $ref: '#/components/schemas/Problem'
+                  $ref: "#/components/schemas/Problem"
+
+  /executions/{id}/artifacts:
+    get:
+      parameters:
+        - in: path
+          name: id
+          schema:
+            type: string
+          required: true
+          description: ID of the script execution
+      tags:
+        - artifacts
+        - executions
+        - api
+      summary: "Get execution's artifacts by ID"
+      description: "Returns arifacts of the given executionID"
+      operationId: getExecutionArtifacts
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Artifact"
+        500:
+          description: "problem with getting execution's artifacts from storage"
+          content:
+            application/problem+json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Problem"
+
+  /executions/{id}/artifacts/{filename}:
+    get:
+      parameters:
+        - in: path
+          name: id
+          schema:
+            type: string
+          required: true
+          description: ID of the script execution
+        - in: path # minio/[bucket_exec_id]/cypress/video/somevideo.avi
+          name: filename
+          schema:
+            type: string
+          required: true
+          description: name of the file in the bucket
+      operationId: downloadFile
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
```

Comment:
see comment further below - we could have a separate response with content type application/octet-stream - see https://stackoverflow.com/a/53339910/249414 for an example - the client would need to specify an accept header to control which response to return